### PR TITLE
Implement GraphQL

### PR DESCRIPTION
**Pull request by drjonah**
Updated ./backend/scraper 
- wrote out script.ts to query data using GraphQL from _ratemyprofessor.com_
- various files added to support script.ts (const.ts, interface.ts, query.ts)

### CODE REVIEW

File: scraper/scraper.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+};
+
+// returns a professor's data based on an id
+// returned Promise with firstName, lastNam'e, id, legacyId, avgDifficulty, avgRating, numRatings
+// example getProfessorData('VGVhY2hlci0yMzM0Nzcy')
+const getProfessorData = async (id: string): Promise<ProfessorPage[]> => {
+  const professorSearchData = await graphQLClient.request(
+    professorRatingQuery,
+    {
+      id: id,
```
Comment (ZombiMigz) [inline scraper/scraper.ts]: update to use shorthand please

**Review by Leaversa — COMMENTED** [review]
Could you include some comments on what each category relates to? I'm not sure if WouldTakeAgain% is added or what legacyID is?

**Review by Leaversa — APPROVED** [review]
Everything looks good! Thanks for the comments and clarification.

**Review by ZombiMigz — APPROVED** [review]
Fantastic work, good job!

**Pull request merged**